samcli/commands/init/__init__.py

```diff
@@ -190,7 +190,7 @@ def _get_cookiecutter_template_context(name, runtime, extra_context):
         try:
             extra_context_dict = json.loads(extra_context)
         except JSONDecodeError:
-            raise UserException(
+            raise click.UsageError(
```

Comment:
How is raising click UsageError different from UserException? I mean what'll be difference on user side due to this change?

Reply:
see https://github.com/awslabs/aws-sam-cli/pull/1686#discussion_r360044936